drivers/SmartThings/zigbee-switch/src/aqara/init.lua

```diff
@@ -70,140 +116,251 @@ local function is_aqara_products(opts, driver, device)
-  device:send(OnOff.attributes.OnOff:read(device))
-  device:send(ElectricalMeasurement.attributes.ActivePower:read(device))
-  device:send(SimpleMetering.attributes.CurrentSummationDelivered:read(device))
+  device:refresh()
+
+  if device:supports_capability(capabilities.energyMeter) or device:supports_capability(capabilities.powerMeter) then
+    device:send(AnalogInput.attributes.PresentValue:read(device):to_endpoint(POWER_METER_ENDPOINT))
+    device:send(AnalogInput.attributes.PresentValue:read(device):to_endpoint(ENERGY_METER_ENDPOINT))
+  end
 end
 
 local function device_info_changed(driver, device, event, args)
-  if device.preferences ~= nil then
-    if device.preferences[MAX_POWER_ID] ~= nil and
-        device.preferences[MAX_POWER_ID] ~= args.old_st_store.preferences[MAX_POWER_ID] then
-      local value = tonumber(device.preferences[MAX_POWER_ID])
-      device:send(cluster_base.write_manufacturer_specific_attribute(device, PREF_CLUSTER_ID, PREF_MAX_POWER_ATTR_ID,
-        MFG_CODE, data_types.SinglePrecisionFloat, max_power_data_type_table[value]))
+  local preferences = device.preferences
+  local old_preferences = args.old_st_store.preferences
+  if preferences ~= nil then
+    for id, attr in pairs(preferences_map) do
+      local old_value = old_preferences[id]
+      local value = preferences[id]
+      if value ~= nil and value ~= old_value then
+        if attr.value_map ~= nil then
+          value = attr.value_map[value]
+        end
+        device:send(cluster_base.write_manufacturer_specific_attribute(device, attr.cluster_id, attr.attribute_id,
+          attr.mfg_code, attr.data_type, value))
+      end
+    end
+  end
+end
+
+local function device_added(driver, device)
+  device:emit_event(capabilities.switch.switch.off())
+
+  -- Only create children for the actual Zigbee device and not the children
+  if device.network_type == device_lib.NETWORK_TYPE_ZIGBEE then
+    local children_amount = get_children_amount(device)
+    if children_amount >= 2 then
+      for i = 2, children_amount, 1 do
+        local name = string.format("%s%d", string.sub(device.label, 0, -2), i)
+        local profile = get_profile_name(device)
+        local metadata = {
+          type = "EDGE_CHILD",
+          label = name,
+          profile = profile,
+          parent_device_id = device.id,
+          parent_assigned_child_key = string.format("%02X", i),
+          vendor_provided_label = name
+        }
+        driver:try_create_device(metadata)
+      end
     end
 
-    if device.preferences[RESTORE_STATE_ID] ~= nil and
-        device.preferences[RESTORE_STATE_ID] ~= args.old_st_store.preferences[RESTORE_STATE_ID] then
-      device:send(cluster_base.write_manufacturer_specific_attribute(device, PREF_CLUSTER_ID,
-        PREF_RESTORE_STATE_ATTR_ID, MFG_CODE, data_types.Boolean, device.preferences[RESTORE_STATE_ID]))
+    -- for wireless button
+    if device:supports_capability(capabilities.button) then
```

Comment:
`supports_capability` checks are also a good indicator that we'd prefer a sub-driver, for readability's sake

Reply:
Moved multi-switch's code to the sub-driver.